src/instrumentation/health_test.go

```diff
@@ -1,77 +1,365 @@
+			fnHealthCheck: fakeHealthCheck(func(ctx context.Context, url string) (health Health, err error) {
+				logger := log.FromContext(ctx)
+				logger.Info("fake health check")
+				return Health{}, nil
+			}),
+			fnInstrumentationStatusUpdater: fakeUpdateInstrumentationStatus(func(ctx context.Context, instrumentation *v1alpha2.Instrumentation) error {
+				logger := log.FromContext(ctx)
+				logger.Info("fake instrumentation status updater")
+				return nil
+			}),
+			namespaces: map[string]*corev1.Namespace{
+				"default":  {ObjectMeta: metav1.ObjectMeta{Name: "default"}},
+				"newrelic": {ObjectMeta: metav1.ObjectMeta{Name: "newrelic"}},
+			},
+			pods: map[string]*corev1.Pod{
+				"default/pod0": {ObjectMeta: metav1.ObjectMeta{Name: "pod0", Namespace: "default", Annotations: map[string]string{"newrelic.com/apm-health": "true"}}},
+			},
+			instrumentations: map[string]*v1alpha2.Instrumentation{
+				"newrelic/instrumentation0": {ObjectMeta: metav1.ObjectMeta{Name: "instrumentation0", Namespace: "newrelic"}},
+			},
+			expectedInstrumentationStatus: v1alpha2.InstrumentationStatus{
+				PodsMatching: 1,
+				PodsNotReady: 1,
+				PodsOutdated: 1,
+				PodsInjected: 1,
+			},
+		},
+		{
+			name: "matching and injected but not ready",
+			fnHealthCheck: fakeHealthCheck(func(ctx context.Context, url string) (health Health, err error) {
+				logger := log.FromContext(ctx)
+				logger.Info("fake health check")
+				return Health{}, nil
+			}),
+			fnInstrumentationStatusUpdater: fakeUpdateInstrumentationStatus(func(ctx context.Context, instrumentation *v1alpha2.Instrumentation) error {
+				logger := log.FromContext(ctx)
+				logger.Info("fake instrumentation status updater")
+				return nil
+			}),
+			namespaces: map[string]*corev1.Namespace{
+				"default":  {ObjectMeta: metav1.ObjectMeta{Name: "default"}},
+				"newrelic": {ObjectMeta: metav1.ObjectMeta{Name: "newrelic"}},
+			},
+			pods: map[string]*corev1.Pod{
+				"default/pod0": {ObjectMeta: metav1.ObjectMeta{Name: "pod0", Namespace: "default", Annotations: map[string]string{
+					"newrelic.com/apm-health":        "true",
+					instrumentationVersionAnnotation: `{"newrelic/instrumentation0":"01234567-89ab-cdef-0123-456789abcdef/55"}`,
+				}}},
+			},
+			instrumentations: map[string]*v1alpha2.Instrumentation{
+				"newrelic/instrumentation0": {ObjectMeta: metav1.ObjectMeta{Name: "instrumentation0", Namespace: "newrelic", UID: "01234567-89ab-cdef-0123-456789abcdef", Generation: 55}},
+			},
+			expectedInstrumentationStatus: v1alpha2.InstrumentationStatus{
+				PodsMatching: 1,
+				PodsNotReady: 1,
+				PodsInjected: 1,
+			},
+		},
+		{
+			name: "matching, injected and ready, but missing health sidecar",
+			fnHealthCheck: fakeHealthCheck(func(ctx context.Context, url string) (health Health, err error) {
+				logger := log.FromContext(ctx)
+				logger.Info("fake health check")
+				return Health{}, fmt.Errorf("fake health check error, url: %q", url)
+				return Health{}, nil
```

Comment:
There are a couple of duplicate returns in this file

Reply:
fixed